integration-tests/transport/test_zero_limiter.py

```diff
@@ -1,174 +1,199 @@
-        elif i == 1:
-            true_field.interpolate(Constant(tracer_min) + expr_1 + expr_2)
-        else:
-            raise ValueError
+    u0 = stepper.fields("u")
+    D0 = stepper.fields("D")
+    b0 = stepper.fields("b")
+    v0 = stepper.fields("water_vapour")
 
-    # ------------------------------------------------------------------------ #
-    # Velocity profile
-    # ------------------------------------------------------------------------ #
+    _, phi, _ = lonlatr_from_xyz(x[0], x[1], x[2])
+
+    uexpr = xyz_vector_from_lonlatr(u_max*cos(phi), 0, 0, x)
+    g = parameters.g
+    w = Omega*R*u_max + (u_max**2)/2
+    sigma = w/10
+
+    Dexpr = H - (1/g)*(w + sigma)*((sin(phi))**2)
+
+    numerator = theta_0 + sigma*((cos(phi))**2) * ((w + sigma)*(cos(phi))**2 + 2*(phi_0 - w - sigma))
+
+    denominator = phi_0**2 + (w + sigma)**2*(sin(phi))**4 - 2*phi_0*(w + sigma)*(sin(phi))**2
+
+    theta = numerator/denominator
+
+    bexpr = parameters.g * (1 - theta)
+
+    initial_msat = q0/(g*Dexpr) * exp(20*theta)
+    vexpr = (1 - xi) * initial_msat
+
+    u0.project(uexpr)
+    D0.interpolate(Dexpr)
+    b0.interpolate(bexpr)
+    v0.interpolate(vexpr)
+
+    # Set reference profiles
+    Dbar = Function(D0.function_space()).assign(H)
+    bbar = Function(b0.function_space()).interpolate(bexpr)
+    stepper.set_reference_profiles([('D', Dbar), ('b', bbar)])
 
-    psi = Function(Vpsi)
-    u = stepper.fields('u')
-
-    # set up solid body rotation for transport
-    # we do this slightly complicated stream function to make the velocity 0 at edges
-    # thus we avoid odd effects at boundaries
-    xc = Ld / 2
-    zc = Ld / 2
-    r = sqrt((x - xc) ** 2 + (z - zc) ** 2)
-    omega = rotations * 2 * pi / tmax
-    r_out = 9 * Ld / 20
-    r_in = 2 * Ld / 5
-    A = omega * r_in / (2 * (r_in - r_out))
-    B = - omega * r_in * r_out / (r_in - r_out)
-    C = omega * r_in ** 2 * r_out / (r_in - r_out) / 2
-    psi_expr = conditional(r < r_in,
-                           omega * r ** 2 / 2,
-                           conditional(r < r_out,
-                                       A * r ** 2 + B * r + C,
-                                       A * r_out ** 2 + B * r_out + C))
-    psi.interpolate(psi_expr)
-
-    gradperp = lambda v: as_vector([-v.dx(1), v.dx(0)])
-    u.project(gradperp(psi))
-
-    return stepper, tmax, true_field
-
-
-@pytest.mark.parametrize('space', [None, 'equispaced'])
-def test_zero_limiter(tmpdir, space):
-
-    # Setup and run
+    return stepper, tmax
+
+
+def test_without_limiter(tmpdir):
+
+    # Setup and run verison without limiter
+    dirname = str(tmpdir)
+
+    stepper_without_limiter, tmax = setup_zero_limiter(dirname)
+
+    stepper_without_limiter.run(t=0, tmax=tmax)
+
+    cloud_without_limiter = stepper_without_limiter.fields('cloud_water')
+
+    assert cloud_without_limiter.dat.data.min() < 0, "The minimum of cloud is negative"
```

Comment:
Should the assert be something like "The minimum of cloud is already non-negative"?

Reply:
Thanks for pointing that out! Should be fixed now